### PR TITLE
Add CrossGen2 Dockerfiles for the various options we want to perf test

### PR DESCRIPTION
**Pull request by MichaelSimons**
This PR is intended to produce the images we would like to perf test as discussed in https://github.com/dotnet/dotnet-docker/issues/1791.

Related to #1820

### CODE REVIEW

File: 5.0/aspnet/alpine3.11/amd64/composed-composite/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+RUN \
+    # Retrieve ASP.NET
+    wget -O aspnetcore.tar.gz https://dotnetcli.azureedge.net/dotnet/aspnetcore/Runtime/$ASPNET_VERSION/aspnetcore-runtime-$ASPNET_VERSION-linux-musl-x64.tar.gz \
+    &&aspnetcore_sha512='029ed32f8a7dfdb258438177b7fd40c187830db35b3d973dec296d8948f99d30a74269c45d31876c9b4533b8d7b6af65dea319299bd9a7081c6881cf026274b4' \
+    && echo "$aspnetcore_sha512  aspnetcore.tar.gz" | sha512sum -c - \
+    && tar -ozxf aspnetcore.tar.gz ./shared/Microsoft.AspNetCore.App -C ./ \
+    && rm aspnetcore.tar.gz \
+    \
+    # Crossgen .NET Core
+    && /crossgen2/crossgen2 \
```
Comment (MichaelSimons) [inline 5.0/aspnet/alpine3.11/amd64/composed-composite/Dockerfile:30]: @davidwrighton - can you help me out with the correct crossgen2 build args here?  

This is for the following scenario:

> I see a fourth option which is the runtime uses composite images and aspnet does as well, but they do not use a unified composite image, but rather one built locally to the particular image.

**Comment by MichaelSimons** [comment]
I am still working on defining the various SDK images.

File: 5.0/aspnet/alpine3.11/amd64/composed-composite/Dockerfile (hunk 1 above)
Comment (davidwrighton) [reply 5.0/aspnet/alpine3.11/amd64/composed-composite/Dockerfile:30]: You need a -r argument to reference the runtime level bits. So, add an argument like -r /dotnet/shared/Microsoft.NETCore.App/$DOTNET_VERSION/*.dll to the command line.

Comment (davidwrighton) [reply 5.0/aspnet/alpine3.11/amd64/composed-composite/Dockerfile:30]: btw, does the change to publish crossgen2 binaries work? I wasn't able to test the end to end scenario without checking in.

File: 5.0/aspnet/alpine3.11/amd64/composed-composite/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ARG RUNTIME_REPO=mcr.microsoft.com/dotnet/core/runtime
+
+
+# Installer image
+FROM $RUNTIME_DEPS_REPO:5.0-alpine3.11 as installer
+
+ENV DOTNET_VERSION 5.0.0-preview.2.20160.6
+
+# Install crossgen2
+RUN wget -O crossgen2.tar.gz https://dotnetbinaries.blob.core.windows.net/dockerassets/microsoft.netcore.app.crossgen2.linux-musl-x64.$DOTNET_VERSION.tar.gz \
```
Comment (MichaelSimons) [inline 5.0/aspnet/alpine3.11/amd64/composed-composite/Dockerfile:12]: This is a temporary solution I built and uploaded until we get on a build that includes the official crossgen2 artifacts - https://github.com/dotnet/runtime/pull/34623

File: 5.0/aspnet/alpine3.11/amd64/composed-composite/Dockerfile (hunk 1 above)
Comment (MichaelSimons) [reply 5.0/aspnet/alpine3.11/amd64/composed-composite/Dockerfile:27]: Thanks @davidwrighton for the guidance on which args to use.  I did run into an issue with the arg which I am not sure if it is a known issue or something wrong with my usage.  I had issues when -r was specified just before the input-files.  The following Dockerfile snippets illustrate what I mean.

**Broken - error regarding no input-files specified**
```
    # Crossgen .NET Core
    && /crossgen2/crossgen2 \
        --composite --inputbubble -O --Os \
        -o /usr/share/dotnet/shared/Microsoft.AspNetCore.App/$ASPNET_VERSION/Microsoft.AspNetCore.App.SharedFramework.dll \
        -r /usr/share/dotnet/shared/Microsoft.NETCore.App/$DOTNET_VERSION/*.dll \
        /usr/share/dotnet/shared/Microsoft.AspNetCore.App/$ASPNET_VERSION/*.dll
```

**Working**
```
    # Crossgen .NET Core
    && /crossgen2/crossgen2 \
        --composite --inputbubble -O --Os \
        -r /usr/share/dotnet/shared/Microsoft.NETCore.App/$DOTNET_VERSION/*.dll \
        -o /usr/share/dotnet/shared/Microsoft.AspNetCore.App/$ASPNET_VERSION/Microsoft.AspNetCore.App.SharedFramework.dll \
        /usr/share/dotnet/shared/Microsoft.AspNetCore.App/$ASPNET_VERSION/*.dll
```

Comment (MichaelSimons) [reply 5.0/aspnet/alpine3.11/amd64/composed-composite/Dockerfile:27]: >btw, does the change to publish crossgen2 binaries work

I hooked-up the docker unit tests which run an e2e app scenario.  The tests are passing which is encouraging.

**Comment by MichaelSimons** [comment]
I'm going to merge.  Feel free to provide feedback at any time and we can refactor.  This is still evolving.

**Pull request merged**